### PR TITLE
add page embed check and css fix

### PR DESCRIPTION
**Pull request by mdroidian**
@dvargas92495 

before
image

after
image

### CODE REVIEW

File: src/components/ResultsView.tsx [hunk 1]
```diff
@@ -122,15 +122,17 @@ const ResultHeader = ({
   );
 };
 
+
```
Comment (dvargas92495) [inline src/components/ResultsView.tsx:125]: let's 🔥 this whitespace

File: src/components/ResultsView.tsx [hunk 2]
```diff
@@ -122,15 +122,17 @@ const ResultHeader = ({
...
+  const isPage = !!getPageTitleByPageUid(uid) ? "page-embed" : "block-embed"
   const contentRef = useRef(null);
   useEffect(() => {
     window.roamAlphaAPI.ui.components.renderBlock({
       uid,
       el: contentRef.current,
     });
   }, [contentRef]);
-  return <div ref={contentRef} className="roamjs-query-embed" />;
+  return <div ref={contentRef} className={`roamjs-query-embed ${isPage}`} />
```
Comment (dvargas92495) [inline src/components/ResultsView.tsx:135]: nice - let's also display the title of the page somewhere in this markup. a simple `<h1 className="rm-page-title">` within the `div` when `isPage` is true should be sufficient

**Review by dvargas92495 — COMMENTED** [review]
nice! left one bit of feedback

Also - after committing the next bit of changes, can you run `npm version patch` before pushing to this PR? this should set RoamJS Query Builder to a new version number, which will automatically deploy the change to Roam Depot review when merged

File: src/components/ResultsView.tsx (hunk 2 above)
Comment (mdroidian) [reply src/components/ResultsView.tsx:135]: I'm having trouble getting the `h1` to stay.  It seems `renderBlock` keeps nuking it.
I've tried adding it inside the `useEffect` adding a separate `useEffect` as well as in the `return`

Would this require a larger refactor or am I missing something simple?

Comment (mdroidian) [reply src/components/ResultsView.tsx:134]: or we can nest the `ref={contentRef}` ... one sec ...

File: src/components/ResultsView.tsx [hunk 3]
```diff
@@ -123,14 +123,15 @@ const ResultHeader = ({
 };
 
 const CellEmbed = ({ uid }: { uid: string }) => {
+  const isPage = !!getPageTitleByPageUid(uid) ? "page-embed" : "block-embed"
```
Comment (dvargas92495) [inline src/components/ResultsView.tsx:126]: Let's make sure we only call `getPageTitleByPageUid(uid)` once - we want to do Roam data lookups as infrequently as possible:
```typescript
const title = getPageTitleByPageUid(uid);
const isPage = !!title ? "page-embed" : "block-embed";
...
<h1 className="rm-title-display">{title}</h1>
```

File: src/components/ResultsView.tsx (hunk 2 above)
Comment (dvargas92495) [reply src/components/ResultsView.tsx:134]: yup this is how we want to do this - it's fine if the title is not editable.

Let's also add a margin-left to the title and remove the margin-top/bottom

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "query-builder",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
[the remark this answers is not in the document]
Comment (mdroidian) [reply package.json]: How would you like to proceed?

**Pull request merged**